tests/test_lltm_resource_admission.py

```
@@ -0,0 +1,100 @@
+    raise AssertionError("compiled core discovered before LLTM resource admission")
+
+
+def test_response_cell_budget_precedes_dense_marshalling(monkeypatch):
+    """Oversized exact response shapes must fail before float64 allocation or Rust."""
+
+    monkeypatch.setattr(lltm, "_MAX_LLTM_MATRIX_CELLS", 2, raising=False)
+    monkeypatch.setattr(fitstats, "_core_module", _unexpected_core)
+    responses = np.broadcast_to(np.array([[0.0]], dtype=np.float64), (3, 1))
+    q_design = np.array([[1.0]], dtype=np.float64)
+
+    with pytest.raises(ValueError, match="responses.*resource limit"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)